indico_jacow/controllers.py

```diff
@@ -309,3 +315,102 @@ def _process(self, data):
         search_affiliations.bump_version()
         return AffiliationSchema().jsonify(aff)
+
+
+class BrevoAPIMixin:
+    @cached_property
+    def api_instance(self):
+        if not hasattr(self, '_api_instance'):
+            from indico_jacow.plugin import JACOWPlugin
+            configuration_brevo = brevo_python.Configuration()
+            configuration_brevo.api_key['api-key'] = JACOWPlugin.settings.get('brevo_api_key')
+            self._api_instance = brevo_python.ContactsApi(brevo_python.ApiClient(configuration_brevo))
+        return self._api_instance
+
+    def get_contact_info(self, email):
+        try:
+            return self.api_instance.get_contact_info(email).to_dict()
+        except ApiException as e:
+            if e.status == 404:
+                return None
+            raise
+
+    def get_all_lists(self):
+        try:
+            return self.api_instance.get_lists().to_dict()
+        except ApiException as e:
+            raise IndicoError(f'Exception when retrieving Mailing Lists from Brevo: {e.reason}')
+
+    def create_contact(self, email, first_name, last_name, list_ids):
+        contact = brevo_python.CreateContact(
+            email=email,
+            attributes={'FIRSTNAME': first_name, 'LASTNAME': last_name},
+            list_ids=list_ids
+        )
+        return self.api_instance.create_contact(contact).to_dict()
+
+
+class RHMailingLists(BrevoAPIMixin, RHUserBase):
+    def _process(self):
+        valid_contact_ids = set()
+        emails = self.user.all_emails
+        lists = self.get_all_lists()
+
+        for email in emails:
+            if (contact_info := self.get_contact_info(email)):
+                if 'list_ids' in contact_info:
+                    valid_contact_ids.update(contact_info['list_ids'])
+
+        for lst in lists.get('lists', []):
+            lst['subscribed'] = lst['id'] in valid_contact_ids
+
+        mailing_lists = json.dumps(lists)
+        return WPUserMailingLists.render_template('mailing_lists.html', 'mailing_lists', user=self.user,
+                                                  mailing_lists=mailing_lists)
+
+
+class RHMailingListSubscribe(BrevoAPIMixin, RHUserBase):
+    @use_kwargs({
+        'list_name': fields.String(required=True, validate=not_empty),
+        'list_id': fields.Int(required=True, validate=not_empty),
+    })
+    def _process(self, list_name, list_id):
+        email = self.user.email
+        try:
+            if self.get_contact_info(email):
+                response = self.add_contact_to_lists(list_id, email)
+            else:
+                response = self.create_contact(
+                    email=email,
+                    first_name=self.user.first_name,
+                    last_name=self.user.last_name,
+                    list_ids=[list_id])
+            self.user.log(UserLogRealm.user, LogKind.positive, 'Mailing Lists', f'Subscribed to list: {list_name}',
```

Comment:
I don't like the idea of taking the list name from the user-provided data here. Please query it from Brevo instead.